### DISCUSSION TITLE
Running Substrait plan in C++

### DISCUSSION THREAD

**Discussion by drabastomek** [duckdb/duckdb#5562] [Q&A]
Hi all,

I'm aware of the Substrait extension that is available thru SQL, Python and R APIs. However, I have a use case that I'd like to pass Substrait plan in a C++ app.

Any guidance would be greatly appreciated.

Thank you,
-Tom

**Comment by snadukudy** [comment]
We got the below answer from DuckDB Team

------
There is no C substrait API yet. Have you tried accessing it through
the query method of a connection?

e.g.,

auto db = make_unique<DuckDB>(nullptr);
auto conn = make_unique<Connection>(*db);
// check that the connection works
auto result = conn->Query("CALL get_substrait("..."));
